autobatch/autobatch.go

```diff
@@ -164,5 +236,32 @@ func (d *Datastore) Close() error {
 	if err2 != nil {
 		return err2
 	}
+	close(d.exit)
+	close(d.newWrite)
```

Comment:
I think this might be racy if the timer fires and prepares a batch, then this flushes and closes the underlying datastore, then the async writer tries to commit the batch but the underlying DS is now closed

Also this could return before the async writer finishes flushing things, which is surprising.